catalyst/core/runner.py

```diff
@@ -151,6 +151,49 @@ def _get_experiment_components(
 
         return model, criterion, optimizer, scheduler, device
 
+    def _process_callbacks_for_stage(self, callbacks):
+        loggers = utils.process_callbacks(
+            OrderedDict(
+                [
+                    (k, v) for k, v in callbacks.items()
+                    if isinstance(v, LoggerCallback)
+                ]
+            )
+        )
+        callbacks = utils.process_callbacks(
+            OrderedDict(
+                [
+                    (k, v) for k, v in callbacks.items()
+                    if not isinstance(v, LoggerCallback)
+                ]
+            )
+        )
+        self.state.loggers = loggers
+        self.loggers = loggers
+        self.callbacks = callbacks
```

Comment:
it's better to define `self.state.loggers` `self.loggers` `self.callbacks` out of scope of this method